src/olympia/files/utils.py

```diff
@@ -243,22 +243,46 @@ def gecko(self):
         """Return the "applications["gecko"]" part of the manifest."""
         return self.get('applications', {}).get('gecko', {})
 
+    @property
+    def guid(self):
+        return self.gecko.get('id', None)
+
+    @property
+    def strict_max_version(self):
+        return get_simple_version(self.gecko.get('strict_max_version'))
+
+    @property
+    def strict_min_version(self):
+        return get_simple_version(self.gecko.get('strict_min_version'))
+
     def apps(self):
         """Get `AppVersion`s for the application."""
         apps = (
             (amo.FIREFOX, amo.DEFAULT_WEBEXT_MIN_VERSION),
             (amo.ANDROID, amo.DEFAULT_WEBEXT_MIN_VERSION_ANDROID)
         )
 
+        doesnt_support_no_id = (
+            self.strict_min_version and
+            vint(self.strict_min_version) <
+                vint(amo.DEFAULT_WEBEXT_MIN_VERSION_NO_ID)
+        )
+
+        if self.guid is None and doesnt_support_no_id:
+            raise forms.ValidationError(
+                _('You have to support Firefox 48+ in order to omit the GUID.')
```

Comment:
I still prefer mine; "requires a GUID to be set" is wordier and harder to parse compared to "required". Plus I like seeing the version number right after the product name eg. "Firefox below 48" versus "Firefox 47 and below"--I think it's clearer.

🚴